### PR TITLE
Fix: Remove the SAML Artifact Binding Support for all saml profiles

### PR DESCRIPTION
**Pull request by RoopGuron**

### CODE REVIEW

**Comment by sreetummidi** [comment]
Review and merge

File: server/src/main/java/org/cloudfoundry/identity/uaa/home/HomeController.java [hunk 1]
```diff
@@ -140,6 +140,12 @@ public String error404(Model model) {
         return "error";
     }
 
+    @RequestMapping("/error400")
```
Comment (fhanik) [inline server/src/main/java/org/cloudfoundry/identity/uaa/home/HomeController.java]: Is this really part of the SAML PR? This looks like a global change.

File: server/src/main/java/org/cloudfoundry/identity/uaa/provider/saml/LoginSamlEntryPoint.java [hunk 2]
```diff
@@ -36,6 +46,30 @@ public void setProviderDefinitionList(SamlIdentityProviderConfigurator providerD
...
+
+        if (!processFilter(fi.getRequest())) {
+            chain.doFilter(request, response);
+            return;
+        }
+
+        try {
+            commence(fi.getRequest(), fi.getResponse(), null);
+        } catch (ServletException e) {
+            if(e.getRootCause() instanceof MetadataProviderException) {
```
Comment (fhanik) [inline server/src/main/java/org/cloudfoundry/identity/uaa/provider/saml/LoginSamlEntryPoint.java]: Is this the only time a MetadataProviderException is thrown? when the binding is not supported? 

Is it possible that a MetadataProviderException can be thrown for other reasons, at which point maybe we should log the error, at least in debug mode?

File: server/src/main/java/org/cloudfoundry/identity/uaa/provider/saml/SPWebSSOProfileImpl.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package org.cloudfoundry.identity.uaa.provider.saml;
```
Comment (fhanik) [inline server/src/main/java/org/cloudfoundry/identity/uaa/provider/saml/SPWebSSOProfileImpl.java]: missing copyright section above package

File: server/src/test/java/org/cloudfoundry/identity/uaa/provider/saml/idp/ZoneAwareIdpMetadataGeneratorTest.java [hunk 4]
```diff
@@ -20,20 +21,22 @@
...
+import java.util.Arrays;
 import java.util.List;
 
 import static org.cloudfoundry.identity.uaa.provider.saml.ZoneAwareMetadataGeneratorTests.cert1Plain;
 import static org.cloudfoundry.identity.uaa.provider.saml.ZoneAwareMetadataGeneratorTests.cert2Plain;
 import static org.cloudfoundry.identity.uaa.provider.saml.ZoneAwareMetadataGeneratorTests.samlKey1;
 import static org.cloudfoundry.identity.uaa.provider.saml.ZoneAwareMetadataGeneratorTests.samlKey2;
 import static org.cloudfoundry.identity.uaa.provider.saml.idp.SamlTestUtils.getCertificates;
-import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.*;
```
Comment (fhanik) [inline server/src/test/java/org/cloudfoundry/identity/uaa/provider/saml/idp/ZoneAwareIdpMetadataGeneratorTest.java]: Set IDE to not generate `*` imports. That way we know what classes are actually used.

File: uaa/src/main/webapp/WEB-INF/web.xml [hunk 5]
```diff
@@ -82,4 +82,9 @@
         <location>/error404</location>
     </error-page>
 
+    <error-page>
```
Comment (fhanik) [inline uaa/src/main/webapp/WEB-INF/web.xml]: this is a global change. does it really pertain to removal of artifact binding?

File: uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/feature/SamlLoginWithLocalIdpIT.java [hunk 6]
```diff
@@ -616,41 +727,48 @@ public void testCrossZoneSamlIntegration() throws Exception {
...
+            webDriver.findElement(By.name("password")).sendKeys("secr3T");
+            webDriver.findElement(By.xpath("//input[@value='Sign in']")).click();
+            assertThat(webDriver.findElement(By.cssSelector("h1")).getText(), Matchers.containsString("Where to?"));
+            Cookie afterLogin = webDriver.manage().getCookieNamed("JSESSIONID");
+            assertNotNull(afterLogin);
+            assertNotNull(afterLogin.getValue());
+            assertNotEquals(beforeLogin.getValue(), afterLogin.getValue());
+        } catch (Exception e) {
+            Assert.
+            assertTrue("Http-Artifact binding is not supported",e instanceof NoSuchElementException);
```
Comment (fhanik) [inline uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/feature/SamlLoginWithLocalIdpIT.java]: this assertion is wrong. there is no guarantee that NoSuchElementException means artifact binding not supported.

**Pull request merged**